docs/spec/qualifiers.rst

```diff
@@ -66,6 +66,22 @@ implementation (or on the first overload, for stubs)::
        def method(self, x=None):
            ...
 
+The ``@final`` decorator can be combined with previous decorators, like
```

Comment:
Somebody on Discuss brought up that perhaps you should put `@property` above `@final`. Do we need to specify that explicitly?

For `classmethod` + `final`, currently both [mypy](https://mypy-play.net/?mypy=latest&python=3.12&gist=b97323c411bc7615c2f3fd2307ee2c21) and [pyright](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMrCoCGANgFBkDGJRAzrVAIIBcZU7UAAoSqWx52p1aEAKYwAFmAAm-dtNHACxEgH1CIWjAAU1WgEooAWgB8mFDGZQQ4gK4gUUAIxyorwTXpjJM9zz4cUApK-mq0opRgKNK6JAbGZqiW1nYOUABMrlSeDABC2oz6rIEewt5SsoHByrxqGlqx8abmyTYw9o4AzBQlQl7iFa7VoarhkdGNhs1JVm0dUAAsQA) allow putting the decorators in any order.

My preference would be to make the spec say explicitly that either order is allowed and add conformance tests for it.

Reply:
What is the correct behavior if a property getter is marked final but a setter or deleter is not (or vice versa)? I ask because the different ordering would seem to imply different behaviors. Does the `final` apply to the `property` object or the method that becomes the `fget` callable?

Reply:
I think it's a good point that the spec text here it too specific ("previous" implies an order, should read "other").

On the other hand, I think it gets a bit too precise to enumerate orders. In Pyre (and I suspect Pyright too) the non-conformance arises from attaching the decorator metadata directly to a function type, so a chained decorator that returns a non-function runs into problems.

In terms of typing, the `@property` and `@final` signatures compose in either order, so I think it's clear that the definition is legal in either order.

Reply:
> What is the correct behavior if a property getter is marked final but a setter or deleter is not (or vice versa)? I ask because the different ordering would seem to imply different behaviors. Does the `final` apply to the `property` object or the method that is wrapped by the property?

This is a great question. I don't feel strongly, but one option:

1. The `final`-ness applies to the slot on the class, not the type itself (property descriptor or otherwise)
2. If the final property is a descriptor, then the setter should not be called on that type (either through `p: Final = Descriptor()` or `@final`-decorated descriptor)